driver.go

```diff
@@ -71,6 +72,9 @@ func (d MySQLDriver) Open(dsn string) (driver.Conn, error) {
 
 	// Enable TCP Keepalives on TCP connections
 	if tc, ok := mc.netConn.(*net.TCPConn); ok {
+		timeout := time.Now().Add(mc.cfg.timeout)
+		tc.SetReadDeadline(timeout)
+		tc.SetDeadline(timeout)
```

Comment:
Control the timeout. if connection broken. exit early. then we can reconnect again.